src/Prometheus/Storage/Redis.php

```diff
@@ -63,9 +63,9 @@ public function __construct(array $options = [])
     /**
      * @param \Redis $redis
      * @return self
-     * @throws StorageException
+     * @throws StorageException|\RedisException
      */
-    public static function fromExistingConnection(\Redis $redis): self
+    public static function fromExistingConnection($redis): self
```

Comment:
Why did you remove the explicit type here?

Reply:
yes, in Hyperf program,Redis instance is not \Redis,but functions are same.

Reply:
By looking at the documentation it states hyperf returns a proxy object to `\Redis`: https://hyperf.wiki/3.1/#/en/redis?id=usage

Looks like the [implementation](https://github.com/hyperf/hyperf/blob/efc3a8b1cc4e52cfff875fabb58bb2520e4824b9/src/redis/src/Redis.php), is still typeable by using `\Hyperf\Redis\Redis`. Would you mind adding that type side-by-side `\Redis|\Hyperf\Redis\Redis`?

Union types where added in php 8.0, this library still supports 7.2. @LKaemmerling do you mind dropping support for outdated and unmaintained php versions, too? I'd like to create a PR right away.